adminapi/base.py

```diff
@@ -388,3 +421,13 @@ def _validate_value(attribute_id, value, datatype=None):
         'existing value from {}'
         .format(attribute_id, type(value), datatype)
     )
+
+
+def cast_datatype(value):
+    for datatype, regexp in STR_BASED_DATATYPES:
+        if regexp.match(str(value)):
+            # date constructor is special.
+            if datatype is date:
+                return date(*value.split('-', 2))
+            return datatype(value)
```

Comment:
Which makes one of our tests crash here with an 
AddressValueError: Unexpected '/' in u'10.0.0.0/24'

Reply:
Fixed.

Reply:
Confirmed!